From 1faf0e00587c45d2100a0692990b23979bf98f97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sun, 7 Jan 2024 23:44:31 +0100
Subject: [PATCH] Makefile: Respect $LDFLAGS when linking the final binary

Among other things, this allows building with `LDFLAGS=-static make` to
obtain a fully statically linked binary. Furthermore, many Linux
distributions include distro-specific optimization in LDFLAGS which are,
without this change, not picked up for pdpmake.
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0470131..f2c0c75 100644
--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@ BINDIR = $(PREFIX)/bin
 OBJS = check.o input.o macro.o main.o make.o modtime.o rules.o target.o utils.o
 
 make: $(OBJS)
-	$(CC) -o make $(OBJS)
+	$(CC) $(LDFLAGS) -o make $(OBJS)
 
 $(OBJS): make.h
 
